diff --git a/package/gluon-config-mode-hostname/Makefile b/package/gluon-config-mode-hostname/Makefile
index 6cb77f5..ae0b2f2 100644
--- a/package/gluon-config-mode-hostname/Makefile
+++ b/package/gluon-config-mode-hostname/Makefile
@@ -6,7 +6,7 @@ include ../gluon.mk
 
 define Package/gluon-config-mode-hostname
 	TITLE:=Set the hostname
-	DEPENDS:=+gluon-config-mode-core
+	DEPENDS:=+gluon-config-mode-core +gluon-web-admin
 endef
 
 $(eval $(call BuildPackageGluon,gluon-config-mode-hostname))
diff --git a/package/gluon-config-mode-hostname/i18n/de.po b/package/gluon-config-mode-hostname/i18n/de.po
index fa5a6e0..43ac402 100644
--- a/package/gluon-config-mode-hostname/i18n/de.po
+++ b/package/gluon-config-mode-hostname/i18n/de.po
@@ -20,3 +20,9 @@ msgstr ""
 "Der Knotenname dient ausschließlich zur Identifikation deines Knotens, z.B. "
 "auf einer Knotenkarte. Er hat keinen Einfluss auf den Namen (SSID) des "
 "ausgestrahlten WLANs."
+
+msgid "The node name can be changed in the advanced settings."
+msgstr "Der Knotenname kann in den erweiterten Einstellungen geändert werden."
+
+msgid "Change node name"
+msgstr "Knotennamen ändern"
diff --git a/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/controller/admin/hostname.lua b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/controller/admin/hostname.lua
new file mode 100644
index 0000000..6e59469
--- /dev/null
+++ b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/controller/admin/hostname.lua
@@ -0,0 +1,3 @@
+package 'gluon-config-mode-hostname'
+
+entry({"admin", "hostname"}, model("admin/hostname"), _("Node name"), 5)
diff --git a/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/model/admin/hostname.lua b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/model/admin/hostname.lua
new file mode 100644
index 0000000..4af5d2d
--- /dev/null
+++ b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/model/admin/hostname.lua
@@ -0,0 +1,30 @@
+local pretty_hostname = require 'pretty_hostname'
+local site = require 'gluon.site'
+local util = require 'gluon.util'
+
+local site_i18n = i18n 'gluon-site'
+local help = site_i18n._translate("gluon-config-mode:hostname-help") or translate(
+	'The node name is used solely for identification of your node, e.g. on a '
+	.. 'node map. It does not affect the name (SSID) of the broadcasted WLAN.'
+)
+
+local default_hostname = util.default_hostname()
+
+local f = Form(translate("Node name"), help)
+local s = f:section(Section)
+local o = s:option(Value, "hostname", translate("Node name"))
+o.datatype = 'minlength(1)'
+
+if site.config_mode.hostname.optional(true) then
+	o.optional = true
+	o.placeholder = default_hostname
+end
+
+o.default = pretty_hostname.get(uci)
+
+function o:write(data)
+	pretty_hostname.set(uci, data or default_hostname)
+	uci:commit('system')
+end
+
+return f
diff --git a/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/wizard/0100-hostname.lua b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/wizard/0100-hostname.lua
--- a/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/wizard/0100-hostname.lua
+++ b/package/gluon-config-mode-hostname/luasrc/lib/gluon/config-mode/wizard/0100-hostname.lua
@@ -1,32 +1,11 @@
 return function(form, uci)
 	local pkg_i18n = i18n 'gluon-config-mode-hostname'
-	local site_i18n = i18n 'gluon-site'
-
 	local pretty_hostname = require 'pretty_hostname'
-	local site = require 'gluon.site'
-	local util = require 'gluon.util'

-	local help = site_i18n._translate("gluon-config-mode:hostname-help") or pkg_i18n.translate(
-		'The node name is used solely for identification of your node, e.g. on a '
-		.. 'node map. It does not affect the name (SSID) of the broadcasted WLAN.'
-	)
-	form:section(Section, nil, help)
-
-	local current_hostname = pretty_hostname.get(uci)
-	local default_hostname = util.default_hostname()
-	local configured = uci:get_first('gluon-setup-mode', 'setup_mode', 'configured', false) or
-		(current_hostname ~= default_hostname)
-
-	local s = form:section(Section)
-	local o = s:option(Value, "hostname", pkg_i18n.translate("Node name"))
-	o.datatype = 'minlength(1)'
-	if site.config_mode.hostname.optional(true) then
-		o.optional = true
-		o.placeholder = default_hostname
-	end
-	if configured or site.config_mode.hostname.prefill(true) then
-		o.default = current_hostname
-	end
-
-	function o:write(data)
-		pretty_hostname.set(uci, data or default_hostname)
-		uci:save('system')
-	end
+	local help = pkg_i18n.translate(
+		'The node name can be changed in the advanced settings.'
+	) .. ' <a href="/admin/hostname">'
+		.. pkg_i18n.translate('Change node name') .. '</a>'
+
+	form:section(Section, pretty_hostname.get(uci), help)
 end
